### PR TITLE
Turn repo into a proper pip package

### PR DESCRIPTION
**Pull request by jcb91**
this simplifies a lot of issues we have installing things, _but_ means that the configure_nbextensions step must be separated into a post-install step executed by the user. This branch hasn't yet updated any readme to reflect that.

### CODE REVIEW

**Comment by jcb91** [comment]
(this is a small part of #626)

**Comment by janschulz** [comment]
I'm not so sure about the name of the package: why themysto and not  `juypter-notebook-extensions`: for me, themysto has something to do with themes, but as far as I know there are no themes for the notebook, just new functionality.

Unfortunately, all the other PRs are also basically unreviewable, as all the renames are in these PRs as well :-(

File: install.py [hunk 1]
```diff
@@ -109,7 +109,8 @@ def install_files():
     for name, src in [('python extensions', 'extensions'),
                       ('templates', 'templates'),
                       ('notebook extensions', 'nbextensions')]:
-        destination = os.path.join(data_dir, src)
+        destination = os.path.join(
+            data_dir, os.path.join('src','themysto', src))
```
Comment (janschulz) [inline install.py]: Why "src/themysto/" and not only "themysto/"?

**Comment by janschulz** [comment]
Why "src/themysto/" and not only "themysto/"?

**Comment by jcb91** [comment]
> I'm not so sure about the name of the package: why themysto and not `juypter-notebook-extensions`

Well, I asked in #577 whether anyone had any ideas for names, but got little repsonse on the name front.
I considered something like `jupyter-contrib-nbextensions`, but with efforts on renaming the repo stalling, decided to skip that for now in favour of something snappier.

> for me, themysto has something to do with themes

an unfortunate similarity, I guess. Themysto came from https://en.wikipedia.org/wiki/Themisto or https://en.wikipedia.org/wiki/Themisto_(moon), with the `i -> y` similar to the [calysto project](https://github.com/Calysto/calysto).


> Why "src/themysto/" and not only "themysto/"?

essentially it avoids a bunch of potential pitfalls in testing and packaging - see https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure for details

**Comment by jcb91** [comment]
> all the renames are in these PRs as well :-(

Not sure quite what you mean here? I guess the github interface doesn't render diffs because of the number of changes (and since almost every file was moved, almost every file has changed)? I generally use a local git gui  for looking at changes, so that I can see one commit at a time, rather than having to look at the whole PR at once (sourcetree is my current client of choice). But yeah, it's something of a barrier to review 😟 Any suggestions for how to make it easier?

**Comment by janschulz** [comment]
> essentially it avoids a bunch of potential pitfalls in testing and packaging - see https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure for details

Ah, I was bitten the other way with wheel builds on windows in the mpl repo: https://github.com/matplotlib/matplotlib/blob/master/patched_bdist_wheel.py

> [Github and 400 files changed] Any suggestions for how to make it easier?

Not anything else than discuss that rename first and then iterate on the rest. Aka "Get this PR merged first" :-)

**Comment by jcb91** [comment]
> I also like that I don't have to install my package to test it

Yeah, I can see the appeal of that. But, given the use of tox, testing is done in a new tox install each time anyway :)

**Comment by jcb91** [comment]
> Not anything else than discuss that rename first and then iterate on the rest. Aka "Get this PR merged first" :-)

Yeah, I'd have done it already (just adapting the `install.py` script to use the updated paths) if I had a more concrete package name, but, as you point out, `themysto` is perhaps not the one we want to settle on...

**Comment by juhasch** [comment]
Can we close this one with the configurator split to a separate repo ?

**Comment by jcb91** [comment]
well I still think we need to do something about this issue, but I think it makes sense to open new PR once I've ironed out install issues, so I'll close this.

**Comment by juhasch** [comment]
I agree about the install issues. What is your opinion on installing the extensions as a package ?

**Comment by jcb91** [comment]
Oh, definitely the way forward. I've got a working version in [my `pippify` branch](https://github.com/jcb91/IPython-notebook-extensions/tree/pippify), but I haven't quite finalised the script which installs the nbextensions yet...

**Pull request closed without merging**